### PR TITLE
Fix absolute position if align-item or justify-content is center or flex-end

### LINKED ISSUE #310: Absolute Position with align items center does not work like on web
```html
<div id="absolute_layout_align_items_center" style="height: 100px; flex-grow: 1; align-items: center; justify-content: center;">
	<div style="position: absolute; width: 50px; height: 50px;"></div>
</div>
```

http://jsfiddle.net/s8ksdpre/5/
@emilsjolander

### PR DESCRIPTION
**Pull request by woehrl01**
Fix #310 (```center```)

additionally fixes if ```flex-end``` is set.

### CODE REVIEW

File: gentest/fixtures/YGAbsolutePositionTest.html [hunk 1]
```diff
@@ -24,3 +24,39 @@
...
+
+<div id="absolute_layout_justify_content_center" style="height: 100px; width: 110px; flex-grow: 1; justify-content: center;">
+	<div style="position: absolute; width: 60px; height: 40px;"></div>
+</div>
+
+<div id="absolute_layout_align_items_center" style="height: 100px; width: 110px; flex-grow: 1; align-items: center;">
+	<div style="position: absolute; width: 60px; height: 40px;"></div>
+</div>
+
+<div id="absolute_layout_align_items_center_on_child_only" style="height: 100px; width: 110px; flex-grow: 1;">
```
Comment (emilsjolander) [inline gentest/fixtures/YGAbsolutePositionTest.html]: we have tests for align-self overriding align-items. No need to test this here. Remove the test.

File: yoga/Yoga.c [hunk 2]
```diff
@@ -1384,6 +1392,14 @@ static void YGNodeAbsoluteLayoutChild(const YGNodeRef node,
                                                  child->layout.measuredDimensions[dim[crossAxis]] -
                                                  YGNodeTrailingBorder(node, crossAxis) -
                                                  YGNodeTrailingPosition(child, crossAxis, width);
+  }else if (!YGNodeIsLeadingPosDefined(child, crossAxis) &&
```
Comment (emilsjolander) [inline yoga/Yoga.c]: space before `else`

File: yoga/Yoga.c [hunk 3]
```diff
@@ -1384,6 +1392,14 @@ static void YGNodeAbsoluteLayoutChild(const YGNodeRef node,
                                                  child->layout.measuredDimensions[dim[crossAxis]] -
                                                  YGNodeTrailingBorder(node, crossAxis) -
                                                  YGNodeTrailingPosition(child, crossAxis, width);
+  }else if (!YGNodeIsLeadingPosDefined(child, crossAxis) && 
+    YGNodeAlignItem(node, child) == YGAlignCenter) {
+    child->layout.position[leading[crossAxis]] = (node->layout.measuredDimensions[dim[crossAxis]] -
+                                                  child->layout.measuredDimensions[dim[crossAxis]]) / 2.0f;
+  }else if (!YGNodeIsLeadingPosDefined(child, crossAxis) &&
```
Comment (emilsjolander) [inline yoga/Yoga.c]: space before `else`

File: yoga/Yoga.c [hunk 4]
```diff
@@ -1376,6 +1376,14 @@ static void YGNodeAbsoluteLayoutChild(const YGNodeRef node,
                                                 child->layout.measuredDimensions[dim[mainAxis]] -
                                                 YGNodeTrailingBorder(node, mainAxis) -
                                                 YGNodeTrailingPosition(child, mainAxis, width);
+  }else if (!YGNodeIsLeadingPosDefined(child, mainAxis) &&
```
Comment (emilsjolander) [inline yoga/Yoga.c]: space before `else`

File: yoga/Yoga.c [hunk 5]
```diff
@@ -1376,6 +1376,14 @@ static void YGNodeAbsoluteLayoutChild(const YGNodeRef node,
                                                 child->layout.measuredDimensions[dim[mainAxis]] -
                                                 YGNodeTrailingBorder(node, mainAxis) -
                                                 YGNodeTrailingPosition(child, mainAxis, width);
+  }else if (!YGNodeIsLeadingPosDefined(child, mainAxis) && 
+      node->style.justifyContent == YGJustifyCenter) {
+    child->layout.position[leading[mainAxis]] = (node->layout.measuredDimensions[dim[mainAxis]] -
+                                                 child->layout.measuredDimensions[dim[mainAxis]]) / 2.0f;
+  }else if (!YGNodeIsLeadingPosDefined(child, mainAxis) &&
```
Comment (emilsjolander) [inline yoga/Yoga.c]: space before `else`

File: gentest/fixtures/YGAbsolutePositionTest.html (hunk 1 above)
Comment (woehrl01) [inline gentest/fixtures/YGAbsolutePositionTest.html]: I actually think we need this test, to keep verifing that we call ```YGNodeAlignItem``` and we won't check only on the parent node style. On lines 1396 and 1400.

Comment (emilsjolander) [inline gentest/fixtures/YGAbsolutePositionTest.html]: Yeah ok

**Pull request closed without merging**